lib/plugins/aws/invokeLocal/invoke.rb

```diff
@@ -42,6 +42,11 @@ def log(message)
   end
 end
 
+
+def attach_tty
+  $stdin.reopen "/dev/tty", "a+" unless Gem.win_platform? || $stdin.tty? || !File.exist?("/dev/tty")
```

Comment:
Could you add a call to `tty` if platform is linux? Equivalent python code:

https://github.com/serverless/serverless/blob/f54081216c72cc48b0d91657ccca28be2510d031/lib/plugins/aws/invokeLocal/invoke.py#L76-L79

Reply:
Yes. What is the purpose of doing this though - is it checking that there _is_ a tty terminal?

As an aside, should be ok to do it on os x too by the looks of it:

```
$ tty
/dev/ttys001
$
```

Reply:
I'm not 100% sure, but I think it ensures the TTY is usable. Despite it working like that for you on a mac, the check didn't actually work :(

Reply:
hmm. `system('tty')` returns a 1 on linux? what if you run it at the CLI?

Reply:
Nevermind. Just looked a bit more at various tty stuff in ruby & python. I'm OK with the code as is now. I'm actually gonna change the python check to be more like this one (using language native checks)